src/base/bittorrent/session.cpp

```diff
@@ -4602,7 +4608,7 @@ void Session::handleExternalIPAlert(const lt::external_ip_alert *p)
 
 void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
 {
-    const qreal interval = lt::total_milliseconds(p->timestamp() - m_statsLastTimestamp) / 1000.;
+    const qreal interval = std::max(lt::total_milliseconds(p->timestamp() - m_statsLastTimestamp) / 1000., .01);
```

Comment:
I really dislike this change, care to justify it?

Reply:
@Chocobo1: 

> care to justify it? 

According to the discussion linked by @jagannatharjun, qBT likes to truncate the stats refresh interval to zero. 

I would presume this is a workaround to prevent that from ever happening again.

Reply:
the interval becomes zero when the gap between consecutive calls of `handleSessionAlert` is <= 15 milliseconds and causes "impossible speed graph" bug, Actually, according to our perspective this shouldn't happen i.e there should be at least 150 msec gap between consecutive calls for `handelSessionStatsAlert` but it still happens, so I think it's more of a Qt thing, where the queues get bloated doesn't execute on time on high CPU usage scenario.

Reply:
Can we use appropriate integral type for interval to avoid division here (that can lead to truncation here)?
Also shouldn't we compare last and current timestamps? If they are really equal we should drop current alert.

Reply:
> Also shouldn't we compare last and current timestamps? If they are really equal we should drop current alert.

I would say drop it when the interval is zero because see there is always some gap between call but still interval becomes zero somehow, see the logs of a spike (without any changes, current qbittorrent master branch) (handleSessionStatsAlert - time uploadRate downloadRate)
```
21.07.2020 18:12:39.563        66138                    1791           
21.07.2020 18:12:39.578 9223372036854775808            9223372036854775808           
21.07.2020 18:12:39.985      1190169                   32694     
```

here there are 15 milliseconds between the calls  but still, on second call the interval is zero that's why you see weird value

you can check it yourself set refresh timer's timeout to 15 https://github.com/qbittorrent/qBittorrent/blob/2a638d16e023968ea501f491be508d442fdf723f/src/base/bittorrent/session.cpp#L555

and add assert `assert(interval >0.)` inside handleSessionStatsAlert, the assert will fail

Reply:
>and add assert assert(interval >0.) inside handleSessionStatsAlert, the assert will fail

Just in case you didn't quite understand my previous comment, I suggest that you distinguish between cases when timestamps are really equal, and when there is truncation due to floating-point calculations.
We need to fix the possibility of the second one by rethinking the logic.
As for the possibility of two alerts with really equal timestamps we need to first understand how it's possible. If these alerts provide the same data, we can simply discard the repeated alert.